tests/src/Functional/QueryTestBase.php

```diff
@@ -50,8 +52,20 @@ protected function query($query, array $variables = NULL, $operation = NULL) {
       'operation' => $operation,
     ];
 
+    // Ensure that requests are made in the right session.
+    $minkSession = $this->getSession()->getCookie($this->getSessionName());
+
+    $cookie = new SetCookie();
+    $cookie->setName($this->getSessionName());
+    $cookie->setValue($minkSession);
+    $cookie->setDomain(parse_url($this->baseUrl)['host']);
```

Comment:
Passing `PHP_URL_HOST` as the second parameter to `parse_url()` prevents the need for array access on the return value.